manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -103,10 +102,10 @@ public Processor createErrorHandlerProcessor(String bridgeId, String customerId,
             throw new InternalPlatformException("Unable to configure error handler");
         }
 
-        return doCreateProcessor(bridge, customerId, owner, ProcessorType.ERROR_HANDLER, processorRequest);
+        return doCreateProcessor(bridge, customerId, owner, ProcessorType.ERROR_HANDLER, processorRequest, bridge.getGeneration());
     }
 
-    private Processor doCreateProcessor(Bridge bridge, String customerId, String owner, ProcessorType processorType, ProcessorRequest processorRequest) {
+    private Processor doCreateProcessor(Bridge bridge, String customerId, String owner, ProcessorType processorType, ProcessorRequest processorRequest, long generation) {
```

Comment:
wouldn't be more suitable to have some self explaining enum `GenerationStrategy long` rather than `long generation`?

Reply:
No, it needs to be a numerical type.

`generation` increases each time a `Bridge` is [saved](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/875/files#diff-66557df441afc3933a43c346564ae30380002fb47b8a0f12612ebcf2e71ece79R128). When updating a `Bridge` instance in` BridgeWorker` if the Error Handler `Processor` generation [is not equal to](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/875/files#diff-1319ec61fecd04bc61a3e44b2bffba77efe39a0885cf8d585aa9b01b97736859R122) the that of the `Bridge` it means the `Processor` is out of date and needs updating too. We then [set](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/875/files#diff-58ef592841ec60f16af57955f1413b4962e3111a85558c21e9f346bbf10ae4f9R122) the Error Handler `Processor` generation to be the same as that for the `Bridge.

Reply:
Ok, then it is fair to no use enum. However I do not see any `+ 1` mechanism in `BridgeWorker` as you mentioned in `ConnectorWorker`

Reply:
That's a great question!

I've tried to answer it at least three times; but I am failing! I recall I was setting the Error Handler generation to equal that of the `Bridge` to avoid unnecessary updates that I saw happening when testing; but I cannot remember the exact circumstances. It also leads me to then believe the same should be done in `ConnectorWorker` (set to that of the `Processor`) and not the increment as it does now. I am also concerned setting a `Processor` generation to that of the `Bridge` is wrong for regular processors not _tied_ to the life-cycle of the `Bridge`.

I need to investigate more..

Reply:
https://issues.redhat.com/browse/MGDOBR-975